src/Components/Contact/ContactForm.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from 'react'
+import { Formik, Form, Field, ErrorMessage } from 'formik'
+
+import '../FormStyles.css'
+
+const ContactForm = () => {
+  return (
+    <>
+      <Formik
+        initialValues={{
+          name: '',
+          email: '',
+          phone: '',
+          message: '',
+        }}
+        validate={(values) => {
+          let errors = {}
+          if (!values.name) {
+            errors.name = 'Debes poner tu nombre'
+          }
+
+          if (!values.email) {
+            errors.email = 'Debes poner tu email'
+          } else if (
+            !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
+          ) {
+            errors.email = 'La dirección de Email no es valida'
+          }
+          if (!values.phone) {
+            errors.phone = 'Debes poner tu número de telefono'
+          }
+          if (values.phone.length < 8) {
+            errors.phone = 'El numero de teléfono no es válido'
+          }
+          if (!values.message) {
+            errors.message = 'Debes escribir un mensaje'
+          }
+          return errors
+        }}
+        onSubmit={(values, { resetForm }) => {
+          resetForm()
+        }}
+      >
+        {({ errors }) => (
+          <Form className='form-container'>
+            <label>Nombre</label>
+            <Field
+              className='input-field'
+              type='text'
+              name='name'
+              placeholder='Escribe tu Nombre'
+            />
+            <ErrorMessage
+              name='name'
+              component={() => (
+                <div className='form-text error'>{errors.name}</div>
+              )}
+            />
```

Comment:
because this is being repeted a lot of times maybe can be a separated component or a function like 
```
const showError = (label, name, placeholder, message) => (
            <label>{label}</label>
            <Field
              className='input-field'
              type='text'
              name={name}
              placeholder={placeholder}
            />
            <ErrorMessage
              name='name'
              component={() => (
                <div className='form-text error'>{message}</div>
              )}
            />
)
```

Reply:
other thing that can be done here is a method that iterate over the errors object and show the error for each key. something like:
```
const showAllErrors = (errors) => errors.keys().map( field => {
    const currentErrorMessage = errors[field];
    const placeholder = `Escribe tu ${field}`;  
    return showError(field, field, placeholder, currentErrorMessage);
})
```

Reply:
if you wnat to show a different label from the key you can create a switch to link the label with the message. It could be on a getLabel function or something like that.

Reply:
The last two suggestions are not required. its to start thinking on modularization and how it can be go in a way that a form can grow up without modifying the form.

Reply:
I mixed the error and the input in a new reusable component. It's better now!
I still need to keep refactoring the validation function for not repeating code.